json/src/test/java/io/airlift/json/TestJsonModule.java

```diff
@@ -89,7 +89,6 @@ public void testJsonCodecFactoryBinding()
     public void testSetup()
             throws Exception
     {
-        assertThat(CAR).isEqualTo(CAR);
```

Comment:
That was intentional, to ensure that the `.equals()` method for that object was correct. I guess that was lost when this switched to `AssertJ`, which short-circuits that check internally.

TestNG assertions used to do:

```java
    if ((expected == null) && (actual == null)) {
      return true;
    }
    // Only one of them is null
    if (expected == null || actual == null) {
      return false;
    }
    return expected.equals(actual) && actual.equals(expected);
```